spec/services/one_off_redirect_fix_service_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+RSpec.describe OneOffRedirectFixService do
+  it "should find the first non-redirect edition and update all redirects in the chain" do
+    edition_1 = create(:redirect_live_edition, destination: "/redirected-path-1", base_path: "/original-base-path")
+    edition_2 = create(:redirect_live_edition, destination: "/redirected-path-2", base_path: "/redirected-path-1")
+    edition_3 = create(:redirect_live_edition, destination: "/redirected-path-3", base_path: "/redirected-path-2")
+    edition_4 = create(:live_edition, base_path: "/redirected-path-3")
+
+    described_class.update_redirects(edition_1)
+    described_class.update_redirects(edition_2)
+    described_class.update_redirects(edition_3)
+    expect(edition_1.reload.redirects.first[:destination]).to eq edition_4.base_path
+    expect(edition_2.reload.redirects.first[:destination]).to eq edition_4.base_path
+    expect(edition_3.reload.redirects.first[:destination]).to eq edition_4.base_path
+  end
+
+  it "should not update redirects for English edition" do
+    document = create(:document)
+    edition_1 = create(:redirect_live_edition, destination: "/redirected-path-1", base_path: "/original-base-path", document: document)
+    create(:live_edition, base_path: "/redirected-path-2")
+
+    described_class.fix_redirects!
+    expect(edition_1.reload.redirects.first[:destination]).to eq "/redirected-path-1"
+  end
+
+  it "should skip updating edition before the one and the one with more than 1 redirect" do
```

Comment:
This test looks like we're expecting editions 1-4 to be unchanged at the end of the `update_redirects` calls. (Which is also the behaviour I'd expect). So I'd rephrase the test name to:

```suggestion
  it "should avoid updating editions when any editions in the chain have more than 1 redirect" do
```